swap/src/api/request.rs

```diff
@@ -0,0 +1,834 @@
+            }
+            Method::GetCurrentSwap => {
+                debug_span!(
+                    "method",
+                    name = "GetCurrentSwap",
+                    log_reference_id = field::Empty
+                )
+            }
+            Method::GetSwapInfo { .. } => {
+                debug_span!(
+                    "method",
+                    name = "GetSwapInfo",
+                    log_reference_id = field::Empty
+                )
+            }
+            Method::History => {
+                debug_span!("method", name = "History", log_reference_id = field::Empty)
+            }
+            Method::ListSellers { .. } => {
+                debug_span!(
+                    "method",
+                    name = "ListSellers",
+                    log_reference_id = field::Empty
+                )
+            }
+            Method::MoneroRecovery { .. } => {
+                debug_span!(
+                    "method",
+                    name = "MoneroRecovery",
+                    log_reference_id = field::Empty
+                )
+            }
+            Method::RawHistory => debug_span!(
+                "method",
+                name = "RawHistory",
+                log_reference_id = field::Empty
+            ),
+            Method::StartDaemon { .. } => {
+                debug_span!(
+                    "method",
+                    name = "StartDaemon",
+                    log_reference_id = field::Empty
+                )
+            }
+            Method::SuspendCurrentSwap => {
+                debug_span!(
+                    "method",
+                    name = "SuspendCurrentSwap",
+                    log_reference_id = field::Empty
+                )
+            }
+            Method::WithdrawBtc { .. } => {
+                debug_span!(
+                    "method",
+                    name = "WithdrawBtc",
+                    log_reference_id = field::Empty
+                )
+            }
+        };
+        if let Some(log_reference_id) = log_reference_id {
+            span.record("log_reference_id", &log_reference_id.as_str());
+        }
+        span
+    }
+}
+
+impl Request {
+    pub fn new(cmd: Method) -> Request {
+        Request {
+            cmd,
+            log_reference: None,
+        }
+    }
+
+    pub fn with_id(cmd: Method, id: Option<String>) -> Request {
+        Request {
+            cmd,
+            log_reference: id,
+        }
+    }
+
+    // We pass the outer tracing span down to this function such that it can be passed down to other spawned tokio tasks
+    // This ensures that tasks like the event_loop are all part of the same tracing span
+    async fn handle_cmd(
+        self,
+        context: Arc<Context>,
+    ) -> Result<serde_json::Value> {
+        match self.cmd {
+            Method::SuspendCurrentSwap => {
+                let swap_id = context.swap_lock.get_current_swap_id().await;
+
+                if swap_id.is_some() {
```

Comment:
This if statement can be removed. The called function will already bail if no swap lock is active